postcss/css-compile-variables.js

```diff
@@ -0,0 +1,87 @@
+import { Color } from "color";
+
+let aliasMap;
+let resolvedMap;
+const RE_VARIABLE_VALUE = /var\((--(.+)--(.+)-(.+))\)/;
+
+function getValueFromAlias(alias) {
+  const derivedVariable = aliasMap.get(`--${alias}`);
+  return resolvedMap.get(derivedVariable); // what if we haven't resolved this variable yet?
+}
+
+function resolveDerivedVariable(decl, variables) {
+  const matches = decl.value.match(RE_VARIABLE_VALUE);
+  if (matches) {
+    const [,wholeVariable, baseVariable, operation, argument] = matches;
+    if (!variables[baseVariable]) {
+      // hmm.. baseVariable should be in config..., maybe this is an alias?
+      if (!aliasMap.get(`--${baseVariable}`)) {
+        throw new Error(`Cannot derive from ${baseVariable} because it is neither defined in config nor is it an alias!`);
+      }
+    }
+    switch (operation) {
```

Comment:
this is great for testing, but just to make sure the plan is clear, the idea is that we have some callback that we pass into the plugin to calculate the color value (and list aliases). This will then call the same logic we would run if we load the theme at runtime as discussed.